mifospay/src/main/java/org/mifos/mobilewallet/mifospay/bank/ui/LinkBankAccountScreen.kt

```diff
@@ -0,0 +1,327 @@
+            contentDescription = bank.name,
+            modifier = Modifier.weight(1f)
+        )
+        Text(
+            text = bank.name,
+            style = MaterialTheme.typography.body1.copy(fontSize = 12.sp),
+            modifier = Modifier
+
+                .padding(top = 4.dp)
+        )
+    }
+}
+
+@Composable
+fun BankList(otherBanks: List<Bank>, onSelectBank: (Bank) -> Unit) {
+
+    Box(
+        modifier = Modifier
+            .background(Color.White)
+    ) {
+        Column(
+        ) {
+            otherBanks.forEachIndexed { index, bank ->
+                OtherBankItem(bank = bank, onSelectBank)
+                Divider(modifier = Modifier.padding(horizontal = 16.dp))
+            }
+        }
+    }
+}
+
+@Composable
+fun OtherBankItem(bank: Bank, onSelectBank: (Bank) -> Unit) {
+    Row(
+        horizontalArrangement = Arrangement.Start,
+        verticalAlignment = Alignment.CenterVertically,
+        modifier = Modifier
+            .padding(vertical = 8.dp, horizontal = 10.dp)
+            .fillMaxWidth()
+            .clickable {
+                onSelectBank(bank)
+            }
+    ) {
+        Image(
+            painter = painterResource(id = bank.image),
+            contentDescription = bank.name,
+            modifier = Modifier.size(30.dp)
+        )
+        Spacer(modifier = Modifier.width(16.dp))
+        Text(text = bank.name, style = TextStyle(fontSize = 14.sp))
+    }
+}
+
+
+@Preview(showSystemUi = true, device = "id:pixel_5")
+@Composable
+private fun LinkBankAccountScreenPreview() {
+    MifosTheme {
+        LinkBankAccountScreen(onSelectBank = {
+
+        })
+    }
+}
+
+private fun getPopularBanks(): List<Bank> {
+    val popularBanks = ArrayList<Bank>()
+    popularBanks.add(Bank("RBL Bank", R.drawable.logo_rbl, 0))
+    popularBanks.add(Bank("SBI Bank", R.drawable.logo_sbi, 0))
+    popularBanks.add(Bank("PNB Bank", R.drawable.logo_pnb, 0))
+    popularBanks.add(Bank("HDFC Bank", R.drawable.logo_hdfc, 0))
+    popularBanks.add(Bank("ICICI Bank", R.drawable.logo_icici, 0))
+    popularBanks.add(Bank("AXIS Bank", R.drawable.logo_axis, 0))
+
+    return popularBanks
+}
+
+private fun getAllBanks(context: Context): List<Bank> {
+    val jsonObject: JSONObject
+
+
+    jsonObject = FileUtils.readJson(context, "banks.json")!!
+    val banksList = ArrayList<Bank>()
+    for (i in 0 until jsonObject.getJSONArray("banks").length()) {
+        banksList.add(
+            Bank(
+                (jsonObject.getJSONArray("banks")[i] as String), R.drawable.ic_bank, 1
+            )
+        )
+    }
+    return banksList
+}
+
+fun filterBanks(allBanks: List<Bank>, query: String, filteredBanks: MutableState<List<Bank>>) {
+    val filteredList = allBanks.filter {
+        it.name.contains(query, ignoreCase = true)
+    }
+    if (query.isBlank()) {
+        filteredBanks.value = allBanks
+    } else {
+        filteredBanks.value = filteredList
+
+    }
+}
```

Comment:
all of this logic shouldn't be written here. Please look at `CardScreenViewModel` & and implement the same in this screen.